api.c

```diff
@@ -114,7 +114,7 @@ static int api_handle_req_add_hostfwd(Slirp * slirp, int fd, struct api_ctx *ctx
 		free(fwd);
 		goto finish;
 	}
-	if (inet_pton(AF_INET, host_addr_s, &fwd->host_addr) != 1) {
+	if (host_addr_s == NULL || inet_pton(AF_INET, host_addr_s, &fwd->host_addr) != 1) {
```

Comment:
q: what does `inet_pton` return when `host_addr_s == NULL`?

Reply:
it segfaults :-)

Reply:
I guess that is expected.  Most C functions segfault if you pass NULL (e.g. strcpy)